src/IB/ConstraintIBMethod.cpp

```diff
@@ -1271,17 +1321,18 @@ ConstraintIBMethod::calculateMomentumOfKinematicsVelocity(const int position_han
                 {
                     const int local_idx = node_idx->getLocalPETScIndex();
                     const double* const X = &X_data[local_idx][0];
+                    const Vector &displacement = node_idx->getPeriodicDisplacement(); // HONG ADDED
```

Comment:
```suggestion
                    const Vector &displacement = node_idx->getPeriodicDisplacement();
```
We have better ways of giving credit